### PR TITLE
Vaishaal/competition

### PR DESCRIPTION
**Pull request by Vaishaal**
Lots of stupid stupid software engineering so we can parameter sweep and shit

* Look at run-kitchensinks-featurize-only notebook to see how to cross validate from python land

Turns out this is miserably hard. We also have serialization of our models and functions that emit the tsv necessary for submission 

look at make_submission_output to generate the output file for the competition. 

We should talk now to figure out how to proceed forward.

### CODE REVIEW

File: bin/run-pipeline-yarn.sh [hunk 1]
```diff
@@ -43,21 +43,21 @@ echo "Using SPARK_SUBMIT=$SPARK_SUBMIT"
...
 
+echo CORES $SPARK_EXECUTOR_CORES
+echo NUM EXECUTORS $SPARK_NUM_EXECUTORS
+
 # Set some commonly used config flags on the cluster
 "$SPARK_SUBMIT" \
   --master $MASTER \
   --class $CLASS \
-  --num-executors 32 \
+  --num-executors  $SPARK_NUM_EXECUTORS \
```
Comment (akmorrow13) [inline bin/run-pipeline-yarn.sh]: where do you set these

File: src/main/scala/net/akmorrow13/endive/utils/Window.scala [hunk 2]
```diff
@@ -82,6 +85,14 @@ case class Window(tf: TranscriptionFactors.Value,
   }
 }
 
+
+case class FeaturizedLabeledWindow(labeledWindow: LabeledWindow, features: DenseVector[Double]) {
+
```
Comment (akmorrow13) [inline src/main/scala/net/akmorrow13/endive/utils/Window.scala]: I don't like that we have 2 case classes that do the same thing. I think we should just keep the old one for simplicity. see https://github.com/akmorrow13/endive/blob/a513f5d7977c2278555db18dbf924dbc728f689d/src/main/scala/net/akmorrow13/endive/processing/Preprocess.scala#L403

File: src/main/scala/net/akmorrow13/endive/EndiveConf.scala [hunk 3]
```diff
@@ -32,13 +32,33 @@ class EndiveConf extends Serializable {
...
   @BeanProperty var heldOutCells: Int = 1
 
+  @BeanProperty var gamma: Double = 1.0
+  @BeanProperty var seed: Int = 0
+  @BeanProperty var readFiltersFromDisk: Boolean = false
+
+  @BeanProperty var valDuringSolve: Boolean = true
+  @BeanProperty var writePredictionsToDisk: Boolean = true
+  @BeanProperty var predictionsOutput: String = "/user/vaishaal/tmp"
+  @BeanProperty var featuresOutput: String = "/user/vaishaal/features"
```
Comment (akmorrow13) [inline src/main/scala/net/akmorrow13/endive/EndiveConf.scala]: can you remove this hardcode to your dir

File: bin/run-pipeline-yarn.sh (hunk 1 above)
Comment (Vaishaal) [inline bin/run-pipeline-yarn.sh]: these are environment variables that you can set anywhere, i set them in the python code

File: src/main/scala/net/akmorrow13/endive/EndiveConf.scala (hunk 3 above)
Comment (Vaishaal) [inline src/main/scala/net/akmorrow13/endive/EndiveConf.scala]: sounds good.

File: src/main/scala/net/akmorrow13/endive/utils/Window.scala (hunk 2 above)
Comment (Vaishaal) [inline src/main/scala/net/akmorrow13/endive/utils/Window.scala]: This name makes more sense.

Comment (Vaishaal) [inline src/main/scala/net/akmorrow13/endive/utils/Window.scala]: We don't really use the other case class for anything.

File: bin/run-pipeline-yarn.sh (hunk 1 above)
Comment (akmorrow13) [inline bin/run-pipeline-yarn.sh]: should we modify the bin/ scripts then so they still run with this change

Comment (Vaishaal) [inline bin/run-pipeline-yarn.sh]: we aren't going to really be using the bin/scripts much anyway, but yeah we can.

File: src/main/scala/net/akmorrow13/endive/utils/Window.scala (hunk 2 above)
Comment (Vaishaal) [inline src/main/scala/net/akmorrow13/endive/utils/Window.scala]: i just did a grep, we only use it in deprecated pipelines so this change won't break anything.

Comment (akmorrow13) [inline src/main/scala/net/akmorrow13/endive/utils/Window.scala]: I use it a lot in Kernel Pipeline. Let's keep BaseFeature

Comment (Vaishaal) [inline src/main/scala/net/akmorrow13/endive/utils/Window.scala]: s/BaseFeature/FeaturizedLabeledWindow/gc :)

Comment (Vaishaal) [inline src/main/scala/net/akmorrow13/endive/utils/Window.scala]: you use it exactly 5 times, the format of kernelpipeline will be changed anyway to match the notebook stuff

**Pull request merged**